app/login.go

```diff
@@ -171,35 +148,16 @@ func (c *LoginClient) login(ctx *cli.Context, domain string, audience string, cl
 		}
 	}
 
-	// According to RFC, we should set a default polling interval if not provided.
-	// https://tools.ietf.org/html/draft-ietf-oauth-device-flow-07#section-3.5
-	if codeResp.Interval == 0 {
-		codeResp.Interval = 10
-	}
-
-	// Get access token
-	tokenResp := OAuthTokenResponse{}
-	for len(tokenResp.AccessToken) == 0 {
-		time.Sleep(time.Duration(codeResp.Interval) * time.Second)
-
-		if err := postFormRequest(
-			domainURL.JoinPath("oauth", "token").String(),
-			url.Values{
-				"grant_type":  {"urn:ietf:params:oauth:grant-type:device_code"},
-				"device_code": {codeResp.DeviceCode},
-				"client_id":   {clientID},
-			},
-			&tokenResp,
-		); err != nil {
-			return err
-		}
+	token, err := config.DeviceAccessToken(ctx.Context, resp)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve access token: %w", err)
 	}
+	fmt.Println("Successfully logged in!")
 
-	tokenRespJson, err := FormatJson(tokenResp)
+	tokenRespJson, err := FormatJson(token)
 	if err != nil {
 		return err
 	}
-	fmt.Println("Successfully logged in!")
 
 	// Save token info locally
 	return c.loginService.WriteToConfigFile(getTokenConfigPath(ctx), tokenRespJson)
```

Comment:
Basic refresh token question from me: Is there a way to never write the access token locally and only write the refresh token? In my brain part of the value of refresh token is that we never need to store the access token so it can't be leaked? Or is it that we fetch the the access token once up front and then subsequently send refresh tokens until they expire and need to be fetched again? To me that is less valuable since we will need to send the access token over the wire every 15s anyways (if that's the refresh token expiry)

Reply:
Ah its the opposite - the refresh token is now more valuable than the access token since it has a longer lifetime and can be used in an exchange for a new access token. Ideally we store this information on the OS' keyring, like using this [library](https://github.com/99designs/keyring), to prevent leaking the token locally.

We will need to store the access token somewhere for use with requests, and the expiration should be in the minutes territory (lets say 15m for illustration).

When the access token is past its expiry, the oauth2 library will exchange the refresh token for a new refresh token and access token before making the gRPC request. The refresh token is exchanged as added protection because it prevents refresh token re-use if credential theft ever occurs. Re-using an invalidated refresh token will also invalidate all refresh tokens handed out to the user, its called [refresh token rotation](https://auth0.com/docs/secure/tokens/refresh-tokens/refresh-token-rotation).

I still need to validate all this behavior in a test environment before merging, but with out current setup I did verify this works.